maidr/util/environment.py

```diff
@@ -11,3 +15,66 @@ def is_interactive_shell() -> bool:
             )
         except ImportError:
             return False
+
+    @staticmethod
+    def initialize_llm_secrets() -> str:
+        """Inject the LLM API keys into the MAIDR instance."""
+
+        gemini_api_key = os.getenv("GEMINI_API_KEY")
+        openai_api_key = os.getenv("OPENAI_API_KEY")
+
+        # Default settings for the MAIDR instance
+        settings = {
+            "vol": "0.5",
+            "autoPlayRate": "500",
+            "brailleDisplayLength": "32",
+            "colorSelected": "#03c809",
+            "MIN_FREQUENCY": "200",
+            "MAX_FREQUENCY": "1000",
+            "keypressInterval": "2000",
+            "ariaMode": "assertive",
+            "openAIAuthKey": "",
+            "geminiAuthKey": "",
+            "skillLevel": "basic",
+            "skillLevelOther": "",
+            "LLMModel": "openai",
+            "LLMPreferences": "",
+            "LLMOpenAiMulti": False,
+            "LLMGeminiMulti": False,
+            "autoInitLLM": True,
+        }
+
+        if gemini_api_key is not None and openai_api_key is not None:
+            settings["geminiAuthKey"] = gemini_api_key
+            settings["openAIAuthKey"] = openai_api_key
+            settings["LLMOpenAiMulti"] = True
+            settings["LLMGeminiMulti"] = True
+            settings["LLMModel"] = "multi"
+        elif openai_api_key is not None:
+            settings["LLMOpenAiMulti"] = True
+            settings["openAIAuthKey"] = openai_api_key
+        elif gemini_api_key is not None:
+            settings["LLMGeminiMulti"] = True
+            settings["geminiAuthKey"] = gemini_api_key
+
+        settings_data = json.dumps(settings)
+
+        keys_injection_script = f"""
+            function addKeyValueLocalStorage(iframeId, key, value) {{
+                const iframe = document.getElementById(iframeId);
+                if (iframe && iframe.contentWindow) {{
+                    try {{
+                        iframe.contentWindow.localStorage.setItem(key, value);
+                    }} catch (error) {{
+                        console.error('Error accessing iframe localStorage:', error);
+                    }}
+                }} else {{
+                    console.error('Iframe not found or inaccessible.');
+                }}
+            }}
+            addKeyValueLocalStorage(
+                'maidr-iframe', 'settings_data', JSON.stringify({settings_data})
```

Comment:
Do NOT use the same `iframeId` here. Based on my testing result, the user env variable change does not get picked up dynamically once a setting is stored to this iframe. since you have pinned down the iframeId as "maidr-iframe-id,"the same LLM setting is preserved regardless of user's env variable update. Use a random id just like https://github.com/dakshpokar/py_maidr/blob/2e08b9d7803c7d544e897cc4b063a5af0f473b8b/maidr/core/maidr.py#L145-L149

However, I still suggest you bind this local store handler to maidr object instead of iframe.

Reply:
I have tested this again and the env variable update gets picked up as expected even with this fixed id. However, please use `class` instead of `id` if you need to go with this tweak. For `id`, assign a random unique iframe id if needed.

Reply:
Sure Professor @jooyoungseo, making this change!

Reply:
Professor, @jooyoungseo. I have assigned a unique id to the maidr iframe. I could've used class but it returns an array of matched elements which I have to then loop over to add the key values.